src/net_processing.cpp

```diff
@@ -2781,6 +2803,30 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    if (msg_type == NetMsgType::DISABLETX) {
+        if (pfrom.fSuccessfullyConnected) {
+            // Disconnect peers that send this message after VERACK; this
+            // must be negotiated between VERSION and VERACK.
+            pfrom.fDisconnect = true;
+            return;
+        }
+        if (pfrom.IsFullOutboundConn()) {
+            // If we picked an outbound for tx-relay and it sends us a
+            // disabletx, we want to find another peer.
+            pfrom.fDisconnect = true;
+            return;
+        }
+        if (pfrom.m_tx_relay != nullptr && WITH_LOCK(pfrom.m_tx_relay->cs_filter, return pfrom.m_tx_relay->fRelayTxes)) {
+            // Can't send a DISABLETX message if they didn't turn off tx-relay
+            // in the version message.
```

Comment:
Some doc suggestions; the first line when you receive a BIP number
```diff
+    // BIPXXX defines feature negotiation of disabletx, which must happen
+    // between VERSION and VERACK.
     if (msg_type == NetMsgType::DISABLETX) {
         if (pfrom.fSuccessfullyConnected) {
-            // Disconnect peers that send this message after VERACK; this
-            // must be negotiated between VERSION and VERACK.
+            // Disconnect peers that send a disabletx message after VERACK.
             pfrom.fDisconnect = true;
             return;
         }
         if (pfrom.IsFullOutboundConn()) {
-            // If we picked an outbound for tx-relay and it sends us a
+            // If we picked an outbound for transaction relay and it sends us a
             // disabletx, we want to find another peer.
             pfrom.fDisconnect = true;
             return;
         }
         if (pfrom.m_tx_relay != nullptr && WITH_LOCK(pfrom.m_tx_relay->cs_filter, return pfrom.m_tx_relay->fRelayTxes)) {
-            // Can't send a DISABLETX message if they didn't turn off tx-relay
-            // in the version message.
+            // Can't send a disabletx message if they didn't turn off
+            // transaction relay in the version message.
             pfrom.fDisconnect = true;
```

Reply:
Done.